### PR TITLE
Fix tunnel and process supervision on Windows

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

These are the same changes from #764, with the addition of a fix so that `tunnel start / stop` work on Windows as well. This PR means to enable us to run supervised commands on Windows, so that we can continue to work on the features that depend on it (node / rails serve, etc.).

### WHAT is this pull request doing?

* Replace `fork` with `Process.spawn` which is more cross-platform
* Kill the supervised process before deleting its stderr / stdout file, which is a requirement on Windows
* Adapt the Tunnel so that it works both OS

### CODE REVIEW

**Comment by paulomarg** [comment]
I've added some changes to ensure the Heroku CLI install process works on Windows. It looks like this at the moment:

heroku_new

The gif doesn't include the rest of the command but it worked as expected during my 🎩 .

File: lib/shopify-cli/heroku.rb [hunk 1]
```diff
@@ -82,6 +86,18 @@ def heroku_command
       local_path = File.join(ShopifyCli::CACHE_DIR, 'heroku', 'bin', 'heroku').to_s
       if File.exist?(local_path)
         local_path
+      elsif @ctx.windows?
+        # Check if Heroku exists in the Windows registry and run it from there
+        require 'win32/registry'
+        begin
+          windows_path = Win32::Registry::HKEY_CURRENT_USER.open('SOFTWARE\heroku') do |reg|
```
Comment (tanema) [inline lib/shopify-cli/heroku.rb:93]: WE ARE HITTING THE REGISTRY. THIS IS POD RACING!

File: lib/shopify-cli/process_supervision.rb [hunk 2]
```diff
@@ -56,15 +56,23 @@ def for_ident(identifier)
...
+        ctx = Context.new
+        pid_file = new(identifier)
+
+        # Make sure the file exists and is empty, otherwise Windows fails
+        File.open(pid_file.log_path, 'w') {}
+        pid = Process.spawn(
+          *args,
+          out: pid_file.log_path,
+          err: pid_file.log_path,
+          in: ctx.windows? ? "nul" : "/dev/null",
```
Comment (tanema) [inline lib/shopify-cli/process_supervision.rb:69]: if you are just using the context here maybe we can just do Context.new.windows?

File: lib/shopify-cli/process_supervision.rb [hunk 3]
```diff
@@ -104,24 +113,30 @@ def running?(identifier)
...
+    # #### Parameters
+    #
+    # * `ctx` - the context of this command
+    #
     # #### Returns
     #
     # * `stopped` - [true, false]
     #
-    def stop
+    def stop(ctx)
```
Comment (tanema) [inline lib/shopify-cli/process_supervision.rb:137]: If you are just passing in a default context above, maybe this should just have a default value

File: lib/shopify-cli/heroku.rb (hunk 1 above)
Comment (paulomarg) [reply lib/shopify-cli/heroku.rb:93]: 😱

File: lib/shopify-cli/process_supervision.rb (hunk 3 above)
Comment (paulomarg) [reply lib/shopify-cli/process_supervision.rb:137]: That's true, all of these contexts are only used to check if we're running on Windows (no messages or anything), so we don't have to pass in any objects - let's just go with `Context.new.windows?` as you suggested.

**Pull request merged**